src/main/java/step3/domain/lotto/WinningNumbers.java

```diff
@@ -0,0 +1,100 @@
+package step3.domain.lotto;
+
+import step3.type.WinningType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class WinningNumbers {
+    private static final int INIT_WINNING_COUNT = 0;
+    private static final int NOT_EXISTS_INDEX = -1;
+    private static final String DELIMITER = ",";
+    private final MarkingNumbers winningNumbers;
+    private final LottoNumber bonusBall;
+
+    private WinningNumbers(MarkingNumbers winningNumbers, LottoNumber bonusBall) {
+        this.winningNumbers = winningNumbers;
+        this.bonusBall = bonusBall;
+    }
+
+    public static WinningNumbers of(String string, int bonusBall) {
+        isValid(string);
+        MarkingNumbers splits = stringToObj(string);
+        return new WinningNumbers(splits, LottoNumber.ofBonus(bonusBall, splits));
+    }
+
+    public static WinningNumbers of(Set<LottoNumber> list, int bonusBall) {
+        isValid(list);
+        MarkingNumbers markingNumbers = new MarkingNumbers(list);
+        return new WinningNumbers(markingNumbers, LottoNumber.ofBonus(bonusBall, markingNumbers));
+    }
+
+    private static MarkingNumbers stringToObj(String string) {
+        Set<LottoNumber> collect = Stream.of(string.split(DELIMITER))
+                .map(value -> {
+                    String trim = value.trim();
+                    return new LottoNumber(Integer.parseInt(trim));
+                })
+                .collect(Collectors.toSet());
+        return new MarkingNumbers(collect);
+    }
+
+    public WinningType getWinningStatistics(LottoTicket ticket) {
+        return compareWinningNumber(ticket);
+    }
+
+    public Map<WinningType, Integer> getWinningStatistics(LottoTickets tickets) {
+        Map<WinningType, Integer> winningStatistics = getInitWinningStatisticsMap();
+        tickets.getTickets()
+                .forEach(ticket -> {
+                    WinningType winningType = compareWinningNumber(ticket);
+                    appendWinningCount(winningStatistics, winningType);
+                });
```

Comment:
```suggestion
        // 메서드로 분리
        Map<WinningType, Long> countPerWinningType = tickets.getTickets()
                                            .stream()
                                            .map(this::compareWinningNumber)
                                            .collect(Collectors.groupingBy(Function.identity(),
                                                                                                  Collectors.counting()));
                                            
                                            
        Map<WinningType, Integer> winningStatistics = Arrays.stream(WinningType.values())
                .filter(winningType -> winningType != WinningType.RANK_ETC)
                .collect(Collectors.toMap(Function.identity(),
                        winningType -> Math.toIntExact(countPerWinningType.get(winningType))));
```
위와 같은 방법도 생각해 볼수 있겠네요 😊

Reply:
제가 Stream API를 해당 과정 시작 초기에 백기선님 강의보고 기본만 익힌지라 이런 방법이 있는지는 몰랐네요 ㅎㅎ 어렵네요 ㅠㅠ 😢